python/cudf/cudf/core/frame.py

```diff
@@ -981,6 +981,14 @@ def _shift(self, offset, fill_value=None):
         data = zip(self._column_names, data_columns)
         return self.__class__._from_table(Frame(data, self._index))
 
+    def __array__(self, dtype=None):
+        raise TypeError(
+            "Implicit conversion to a host NumPy array via __array__ is not allowed, \
+            To explicitly construct a GPU array, consider using \
+            cupy.asarray(...)\nTo explicitly construct a \
```

Comment:
`cupy.asarray(DataFrame)` will not work as `DataFrame` doesn't advertise `__cuda_array_interface__` because it's not an array.

Reply:
@galipremsagar pointed out offline that we already have this redefined in DataFrame 🤦‍♂️.